src/utils/index.ts

```diff
@@ -488,11 +488,14 @@ export async function sleep(ms: number = 0) {
     })
 }
 
+// condition could as well return any instead of boolean, could be convenient sometimes if waiting until a value is returned. Maybe change if such use case emerges.
```

Comment:
`boolean` is perfectly fine IMO.
`any` doesn't communicate that we're expecting a `truthy` or `falsey` result, and in the worst case someone just needs to do `return !!value` instead of `return value`

Also note line length too long.

Reply:
Ahh I see, you mean it could return the first/last `truthy` result of calling `condition()`… interesting.

Probably don't want `any` for that, instead you'd infer the type from `condition`, something like:

```typescript
async function until<T>(condition: MaybeAsync<() => T>, …): Promise<T> => {
…
}
```